src/encrypt.c

```diff
@@ -220,22 +226,33 @@ ece_webpush_encrypt_plaintext(
     padLen -= blockPadLen;
 
     // Fill the rest of the block with plaintext.
-    size_t plaintextEnd = plaintextStart + dataPerBlock - blockPadLen;
-    bool plaintextExhausted = false;
-    if (plaintextEnd >= plaintextLen) {
+    size_t maxBlockPlaintextLen = dataPerBlock - blockPadLen;
+    size_t plaintextEnd;
+    if (maxBlockPlaintextLen > plaintextLen - plaintextStart) {
+      // Equivalent to `plaintextStart + maxBlockPlaintextLen > plaintextLen`
+      // without overflow.
       plaintextEnd = plaintextLen;
-      plaintextExhausted = true;
+    } else {
+      plaintextEnd = plaintextStart + maxBlockPlaintextLen;
     }
     size_t blockPlaintextLen = plaintextEnd - plaintextStart;
 
     // The full length of the plaintext and padding.
     size_t blockLen = blockPlaintextLen + blockPadLen;
 
-    size_t ciphertextEnd = ciphertextStart + blockLen + overhead;
-    if (ciphertextEnd > maxCiphertextLen) {
+    // The full length of the encrypted record, including room for the
+    // plaintext, padding, padding delimiter, and auth tag.
+    size_t recordLen = blockLen + overhead;
```

Comment:
So `recordLen` is what `ciphertextEnd` was before. I'm not sure if I can see that we made sure that this doesn't overflow.